src/main/java/edu/harvard/iq/dataverse/DatasetField.java

```diff
@@ -464,11 +466,50 @@ public boolean isRequired() {
             {
                 required = false;
             }
+            
+            //this is needed to enforce required children validation and display on create #11421
+            //set the parent to required only if a child is set to required at the DV level
+            if (this.datasetFieldType.isCompound() && isHasRequiredChildrenDV()) {
+                required = true;
+            }
+           
         }
         
         return required;
     }
     
+    public boolean isHasRequiredChildrenDV() {
+
+        if (hasRequiredChildrenDV == null) {
+            hasRequiredChildrenDV = false;
+        }
+
+        Dataverse dv = getDataverse();
+        while (!dv.isMetadataBlockRoot()) {
+            if (dv.getOwner() == null) {
+                break; // we are at the root; which by defintion is metadata blcok root, regarldess of the value
+            }
+            dv = dv.getOwner();
+        }
+
+        List<DataverseFieldTypeInputLevel> dftilListFirst = dv.getDataverseFieldTypeInputLevels();
+
+        if (getDatasetFieldType().isHasChildren() && (!dftilListFirst.isEmpty())) {
+            for (DatasetFieldType child : getDatasetFieldType().getChildDatasetFieldTypes()) {
+                for (DataverseFieldTypeInputLevel dftilTest : dftilListFirst) {
+                    if (child.equals(dftilTest.getDatasetFieldType())) {
+                        //We only want to get it here if it is required by 
+                        //the DV and not by default at the installation/DB level
+                        if (dftilTest.isRequired() && !child.isRequired()) {
+                            hasRequiredChildrenDV = true;
+                        }
+                    }
+                }
+            }
+        }
+        return hasRequiredChildrenDV;
+    }
+
```

Comment:
Aside from lines 518/519, it looks like this method and the next one are the same. Is there a way to consolidate?

Reply:
The problem here is maintaining the conditionally required child dataset set field type. If the child is required at the installation level (in the db), but it's parent is not required in the db then it is conditionally required. There's no way to make a child dataset field type conditionally required at the dv level, because there's no way to set the parent as required. We are assuming that if a child field is made required at the dv level then the parent must also be required.